tests/drivers/can/api/src/canfd.c

```diff
@@ -445,6 +605,100 @@ ZTEST_USER(canfd, test_set_timing_data_max)
 	zassert_equal(err, 0, "failed to start CAN controller (err %d)", err);
 }
 
+/**
+ * @brief Test that timing values for the data phase that are above the maximum can't be set.
+ */
+ZTEST_USER(canfd, test_set_timing_data_above_max)
+{
+	int err;
+
+	const struct can_timing * max_timing = can_get_timing_data_max(can_dev);
+	struct can_timing test_timing;
+
+	err = can_stop(can_dev);
+	zassert_equal(err, 0, "failed to stop CAN controller (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.sjw += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid sjw was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.prop_seg += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid prop_seg was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.phase_seg1 += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid phase_seg1 was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.phase_seg2 += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid phase_seg2 was accepted (err %d)", err);
+
+	test_timing = *max_timing;
+	test_timing.prescaler += 1;
+	err = can_set_timing_data(can_dev, &test_timing);
+	zassert_equal(err, -ENOTSUP, "invalid prescaler was accepted (err %d)", err);
+
+	err = can_start(can_dev);
+	zassert_equal(err, 0, "failed to start CAN controller (err %d)", err);
```

Comment:
These are not universal tests. Some CAN controller drivers need to redistribute the timing values (maintaining the sample point) due to hardware limitations (e.g. not all CAN controllers have separate prop_seg and phase_seg1 registers).

Reply:
1. These tests cover
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_common.c#L459
and logic in
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_common.c#L368

2. Test checks if error is returned when timing data passed to can_set_timing_data() is outside can_get_timing_data_min() and can_get_timing_data_max() range.
Can_timing structure is the same for all controllers.
https://docs.zephyrproject.org/latest/doxygen/html/structcan__timing.html
I suspect, that CAN controller that doesn't have separate prop_seg and phase_seg1 shall set min and max value for that field to 0. And, in such case, any other value shall be treated as invalid and -ENOTSUP shall be returned.